### PR TITLE
Add mergePreferred, mergePrioritized docs (#25468)

### PR DESCRIPTION
**Pull request by JoelWee**
References #25468

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

File: akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java [hunk 1]
```diff
@@ -150,6 +150,38 @@ void mergeExample() {
...
+    // #mergePrioritized
+    Source<Integer, NotUsed> sourceA = Source.from(Arrays.asList(1, 2, 3, 4));
+    Source<Integer, NotUsed> sourceB = Source.from(Arrays.asList(10, 20, 30, 40));
+
+    sourceA.mergePrioritized(sourceB, 99, 1, false).runWith(Sink.foreach(System.out::print), system);
+    // prints e.g. 1, 10, 2, 3, 4, 20, 30, 40 since both sources have their first element ready and the left source
+    // has higher priority – if both sources have elements ready, sourceA has a 99% chance of being picked next
+    // while sourceB has a 1% chance
+
+    // Priorities must be positive integers
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java:176]: I don't think this error handling needs to be in the example (same for scaladsl)

**Review by raboof — APPROVED** [review]
I think we shouldn't have the try/catch error handling case in the examples, otherwise LGTM

**Comment by raboof** [comment]
(according to jenkins this needs a `akka-stream-tests/test:scalafmt`

**Comment by JoelWee** [comment]
Hi @ignasi35, please could you help to review this? :)

**Comment by ignasi35** [comment]
Hi @JoelWee , I'm on vacation at the moment. Maybe @ennru is already back from his break.

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Pull request merged**